### PR TITLE
TSDB: improve document description on error

### PR DESCRIPTION
**Pull request by nik9000**
When a document fails to index we'd previously return an error with the
description:
```
failed to parse field [a] of type [integer] in document with id 'null'. Preview of field's value: 'not_an_int'
```

That `with id 'null'` thing is kind of unfortunate. The document *will*
have an id. We just don't know it. So now it says:
```
"failed to parse field [a] of type [integer] in a time series document. Preview of field's value: 'not_an_int'"
```

It's be nice to put the `@timestamp` and the `_tsid` if we have it. But
we mostly won't.

### CODE REVIEW

**Comment by nik9000** [comment]
I've added a bunch of reviewers. I think any one is probably ok.

A point of conversation: we could use this infrastructure to improve the error messages generated when we fail to parse documents in standard mode with an auto-generated id. It'd be a little more complicated than this change, but not much frankly.

File: server/src/main/java/org/elasticsearch/index/mapper/SourceToParse.java
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/mapper/SourceToParse.java:59]:
```suggestion
     * case this will be {@code null} if one isn't sent on the request.
```

File: server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java
Comment (romseygeek) [inline server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java:160]: Maybe add a comment here saying that we don't have an ID yet because it's generated from the content of the document?

**Review by romseygeek — APPROVED** [review]
LGTM! I left a couple of nits, but no re-review required.

**Comment by nik9000** [comment]
> LGTM! I left a couple of nits, but no re-review required.

Thanks! I've merged your suggestion and added the comment you mentioned. And I'll let the robot merge it once everything is checked.

File: server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java
Comment (imotov) [inline server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java:166]: I think specifying a timestamp here might be really helpful.

**Review by imotov — APPROVED** [review]
LGTM, but I think if we could at least extract the timestamp, it would have help to significantly reduce the scope of documents that you need to potentially check.

**Pull request merged**
